File: src/main/java/com/automation/center/config/IConfig.java
package com.automation.center.config;

import org.aeonbits.owner.Config;
import org.aeonbits.owner.Config.Key;
import org.aeonbits.owner.Config.LoadPolicy;
import org.aeonbits.owner.Config.Sources;
import org.aeonbits.owner.ConfigCache;

@LoadPolicy(Config.LoadType.MERGE)
@Sources({"system:properties", "system:env", "classpath:config.properties"})
public interface IConfig extends Config {
    static IConfig getConfig() {
        return ConfigCache.getOrCreate(IConfig.class);
    }

    @Key("lighthouse.url")
    String lighthouseUrl();

    @Key("automation.center.gateway.base.url")
    String gatewayBaseUrl();

    @Key("suite.id")
    long suiteId();
}

File: lighthouse/src/main/java/com/automation/center/config/IConfig.kt
package com.automation.center.config

import org.aeonbits.owner.Config
import org.aeonbits.owner.Config.Key
import org.aeonbits.owner.Config.LoadPolicy
import org.aeonbits.owner.Config.Sources
import org.aeonbits.owner.ConfigCache


@LoadPolicy(Config.LoadType.MERGE)
@Sources("system:properties", "system:env", "classpath:config.properties")
interface IConfig: Config {
    companion object {
        fun getConfig(): IConfig {
            return ConfigCache.getOrCreate(IConfig::class.java)
        }
    }

    @Key("lighthouse.url")
    fun lighthouseUrl(): String

    @Key("automation.center.gateway.base.url")
    fun gatewayBaseUrl(): String

    @Key("suite.id")
    fun suiteId(): Long
}